disnake/i18n.py

```diff
@@ -143,7 +143,7 @@ def __init__(
         key: str = MISSING,
         data: Union[Optional[LocalizationsDict], LocalizationValue] = MISSING,
     ):
-        self.string: StringT = string  # type: ignore  # default values for generic parameters don't quite work
```

Comment:
how did this get solved?

Reply:
with `strictParameterNoneValue` disabled, pyright infers the type of `string: StringT = None` to be `Optional[StringT]`, even though `None` is already assignable to `StringT` since one of the bounds is `Optional[str]`.
With the rule enabled, it gets correctly inferred as just `StringT` which fixes the typing in the assignment here.